### ISSUE TITLE
DockerD Hang/Freeze on MacOS 13.4

### ISSUE THREAD

**Issue by distortedsignal** [docker/for-mac#6976]
### Description

I've been starting/running a lot of containers recently and the Docker daemon on my Mac seems to leak... something? It's fine for the first while, and then it either refuses to schedule containers or it hangs, not responding to requests from the Docker Desktop client or `docker` command line client. Restarting the docker daemon seems to fix the issue. Unfortunately, this behavior is not consistent, and I cannot reliably reproduce the issue.

### Reproduce

The issue is not consistent and cannot be reliably reproduced. However, after the issue has been recreated...

1. `docker image ls` does not display docker images, and does not return

### Expected behavior

`docker image ls` should always display images and should always return

### docker version

```bash
Client:
 Cloud integration: v1.0.35-desktop+001
 Version:           24.0.5
 API version:       1.43
 Go version:        go1.20.6
 Git commit:        ced0996
 Built:             Fri Jul 21 20:32:30 2023
 OS/Arch:           darwin/amd64
 Context:           desktop-linux

Server: Docker Desktop 4.22.1 (118664)
 Engine:
  Version:          24.0.5
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.6
  Git commit:       a61e2b4
  Built:            Fri Jul 21 20:35:45 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.21
  GitCommit:        3dce8eb055cbb6872793272b4f20ed16117344f8
 runc:
  Version:          1.1.7
  GitCommit:        v1.1.7-0-g860f061
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    24.0.5
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.1
    Path:     /Users/tom/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.20.2-desktop.1
    Path:     /Users/tom/.docker/cli-plugins/docker-compose
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /Users/tom/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     /Users/tom/.docker/cli-plugins/docker-extension
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.6
    Path:     /Users/tom/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/tom/.docker/cli-plugins/docker-sbom
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     /Users/tom/.docker/cli-plugins/docker-scan
  scout: Command line tool for Docker Scout (Docker Inc.)
    Version:  0.20.0
    Path:     /Users/tom/.docker/cli-plugins/docker-scout

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 24
 Server Version: 24.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 5.15.49-linuxkit-pr
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 15.63GiB
 Name: docker-desktop
 ID: 7d99c8cb-fc64-4c90-a6bd-d6a19a5c3b1f
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

EA6A9D0F-12E1-4F03-B4D4-915B28957161/20230912215011

### Additional Info

Additional info in thread below...

**Comment by distortedsignal** [comment]
I've seen three modes of failure on this:

1. When using the `docker run` command, containers are created, but they are never scheduled - ie, they never enter the `Running` state.
2. When using the `docker run` command, the container is created and scheduled (it enters the `Running` state), but it gets 0% CPU time (and, thus, cannot perform any meaningful work).
3. The docker daemon fails to respond to any commands at all, causing the command line client to hang and the Docker Desktop application show no data at all (no running containers, no images, etc.)

All of these are fixed by a restart of the docker daemon.

**Comment by ncouse** [comment]
I have seen similar behaviour recently, also on latest 4.22.1.

Everything is running fine, and then containers fail to be created. `docker ps` just hangs, and Docker Desktop dashboard shows no containers. A restart via Docker Desktop solves this.

I didn't spend much investigating it as needed a working system. I would say it happened maybe 3 times in the last week for me. I am constanty building, running, tearing down containers all day.

**Comment by distortedsignal** [comment]
Docker team! Let me know if there's anything I can do to help. I know this ticket is sparse and kinda not great, and I'd love to help get this resolved 🙂

**Comment by distortedsignal** [comment]
Ok, I just experienced a Type 1 failure and I got the Diagnostics ID `EA6A9D0F-12E1-4F03-B4D4-915B28957161/20230913164359` from the report that I generated from it. As a reminder, this is where the containers are Created (ie: they have the `Created` status when I `docker ps -a`) but they are never scheduled (they never enter the `Running` state despite being created with `docker run`).

**Comment by distortedsignal** [comment]
Looking at the `dockerd.log`, we're ignoring a lot of `TaskDelete` events:

<details><summary>dockerd.log</summary>

Raw log:

```
[2023-09-13T17:17:27.712524789Z][dockerd][I] time="2023-09-13T17:17:27.711289620Z" level=info msg="ignoring event" container=e661ab7515da6f628155cec081a40c47655c1ccc5b5c2f1dcefd896dd25878bf module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
[2023-09-13T17:18:16.124670301Z][dockerd][I] time="2023-09-13T17:18:16.123418862Z" level=info msg="ignoring event" container=3921503330fcf2b1899ae936ee3b12632baad4d380dc97dfc6e7e1b55baf58e2 module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
[2023-09-13T17:19:03.468211658Z][dockerd][I] time="2023-09-13T17:19:03.467077968Z" level=info msg="ignoring event" container=a560055cd41102c3f31260a17a82c05959bd31d9a01c05e704e6e11e0401b126 module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
[2023-09-13T17:19:56.995204441Z][dockerd][I] time="2023-09-13T17:19:56.994983822Z" level=info msg="ignoring event" container=beb7cdad0a2f3db43a5460f2979b60f8c6cccb98f86b556c8d0ba407d32bc85b module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
[2023-09-13T17:20:47.171656189Z][dockerd][I] time="2023-09-13T17:20:47.170299060Z" level=info msg="ignoring event" container=2b97372a69715896d15c2dc3470e88ff96fb61fdbb5bc6048b141705b3b80493 module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
[2023-09-13T17:21:34.055665387Z][dockerd][I] time="2023-09-13T17:21:34.055003808Z" level=info msg="ignoring event" container=1b555b63533a1cc8d27ea36a975bab258279bf3a11a1364cb530cf42ab3344ba module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
```

Readable log (times approximate):

```
[T0]     [dockerd][I] time="T0"      level=info msg="ignoring event" container=sha:e661... module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
[T+45s]  [dockerd][I] time="T+45s"   level=info msg="ignoring event" container=sha:3921... module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
[T+1m30s][dockerd][I] time="T+1m30s" level=info msg="ignoring event" container=sha:a560... module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
[T+2m30s][dockerd][I] time="T+2m30s" level=info msg="ignoring event" container=sha:beb7... module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
[T+3m20s][dockerd][I] time="T+3m20s" level=info msg="ignoring event" container=sha:2b97... module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
[T+4m5s] [dockerd][I] time="T+4m5s"  level=info msg="ignoring event" container=sha:1b55... module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
```

</details>

But we're also loading a bunch of plugins, starting some kind of loop, disconnecting a shim, and then cleaning all of that up:

<details><summary>containerd.log</summary>

Raw log:

```
[2023-09-13T17:24:48.624290207Z][containerd][I] time="2023-09-13T17:24:48.622776405Z" level=info msg="loading plugin \"io.containerd.event.v1.publisher\"..." runtime=io.containerd.runc.v2 type=io.containerd.event.v1
[2023-09-13T17:24:48.624384970Z][containerd][I] time="2023-09-13T17:24:48.622910356Z" level=info msg="loading plugin \"io.containerd.internal.v1.shutdown\"..." runtime=io.containerd.runc.v2 type=io.containerd.internal.v1
[2023-09-13T17:24:48.624414711Z][containerd][I] time="2023-09-13T17:24:48.622935717Z" level=info msg="loading plugin \"io.containerd.ttrpc.v1.task\"..." runtime=io.containerd.runc.v2 type=io.containerd.ttrpc.v1
[2023-09-13T17:24:48.626583502Z][containerd][I] time="2023-09-13T17:24:48.623987692Z" level=info msg="starting signal loop" namespace=moby path=/var/run/desktop-containerd/daemon/io.containerd.runtime.v2.task/moby/a723f0edc92d0b43ea1cc2734d9191364497aee0cba7ff709503bf5eaf78b205 pid=17402 runtime=io.containerd.runc.v2
[2023-09-13T17:25:29.611541737Z][containerd][I] time="2023-09-13T17:25:29.610830201Z" level=info msg="shim disconnected" id=a723f0edc92d0b43ea1cc2734d9191364497aee0cba7ff709503bf5eaf78b205
[2023-09-13T17:25:29.611627057Z][containerd][I] time="2023-09-13T17:25:29.611158159Z" level=warning msg="cleaning up after shim disconnected" id=a723f0edc92d0b43ea1cc2734d9191364497aee0cba7ff709503bf5eaf78b205 namespace=moby
[2023-09-13T17:25:29.611652606Z][containerd][I] time="2023-09-13T17:25:29.611186082Z" level=info msg="cleaning up dead shim"
[2023-09-13T17:25:29.646968434Z][containerd][I] time="2023-09-13T17:25:29.646471472Z" level=warning msg="cleanup warnings time=\"2023-09-13T17:25:29Z\" level=info msg=\"starting signal loop\" namespace=moby pid=17556 runtime=io.containerd.runc.v2\n"
[2023-09-13T17:25:33.789856613Z][containerd][I] time="2023-09-13T17:25:33.788725901Z" level=info msg="loading plugin \"io.containerd.event.v1.publisher\"..." runtime=io.containerd.runc.v2 type=io.containerd.event.v1
[2023-09-13T17:25:33.789933003Z][containerd][I] time="2023-09-13T17:25:33.789013671Z" level=info msg="loading plugin \"io.containerd.internal.v1.shutdown\"..." runtime=io.containerd.runc.v2 type=io.containerd.internal.v1
[2023-09-13T17:25:33.789965175Z][containerd][I] time="2023-09-13T17:25:33.789163661Z" level=info msg="loading plugin \"io.containerd.ttrpc.v1.task\"..." runtime=io.containerd.runc.v2 type=io.containerd.ttrpc.v1
[2023-09-13T17:25:33.791955640Z][containerd][I] time="2023-09-13T17:25:33.790858303Z" level=info msg="starting signal loop" namespace=moby path=/var/run/desktop-containerd/daemon/io.containerd.runtime.v2.task/moby/f3d97f992cce3d53ab7a03f44dc1bb3b51d5ee1720e92cc95ef871088ec22b81 pid=17717 runtime=io.containerd.runc.v2
[2023-09-13T17:25:42.944167883Z][containerd][I] time="2023-09-13T17:25:42.942346194Z" level=info msg="loading plugin \"io.containerd.event.v1.publisher\"..." runtime=io.containerd.runc.v2 type=io.containerd.event.v1
[2023-09-13T17:25:42.944485699Z][containerd][I] time="2023-09-13T17:25:42.942646595Z" level=info msg="loading plugin \"io.containerd.internal.v1.shutdown\"..." runtime=io.containerd.runc.v2 type=io.containerd.internal.v1
[2023-09-13T17:25:42.944531453Z][containerd][I] time="2023-09-13T17:25:42.942676997Z" level=info msg="loading plugin \"io.containerd.ttrpc.v1.task\"..." runtime=io.containerd.runc.v2 type=io.containerd.ttrpc.v1
[2023-09-13T17:25:42.944543862Z][containerd][I] time="2023-09-13T17:25:42.943620173Z" level=info msg="starting signal loop" namespace=moby path=/var/run/desktop-containerd/daemon/io.containerd.runtime.v2.task/moby/d6bebdff2d1ae0080b8d9ee369cf36f2c7bb142246927146a91062e95e722a15 pid=17929 runtime=io.containerd.runc.v2
[2023-09-13T17:26:17.892438972Z][containerd][I] time="2023-09-13T17:26:17.891185581Z" level=info msg="shim disconnected" id=f3d97f992cce3d53ab7a03f44dc1bb3b51d5ee1720e92cc95ef871088ec22b81
[2023-09-13T17:26:17.892505957Z][containerd][I] time="2023-09-13T17:26:17.891518885Z" level=warning msg="cleaning up after shim disconnected" id=f3d97f992cce3d53ab7a03f44dc1bb3b51d5ee1720e92cc95ef871088ec22b81 namespace=moby
[2023-09-13T17:26:17.893361494Z][containerd][I] time="2023-09-13T17:26:17.891559591Z" level=info msg="cleaning up dead shim"
[2023-09-13T17:26:17.935879450Z][containerd][I] time="2023-09-13T17:26:17.934491610Z" level=warning msg="cleanup warnings time=\"2023-09-13T17:26:17Z\" level=info msg=\"starting signal loop\" namespace=moby pid=18062 runtime=io.containerd.runc.v2\n"
```

Readable log (times approximate):

```
[T0]         [containerd][I] time="T0"          level=info msg="loading plugin \"io.containerd.event.v1.publisher\"..." runtime=io.containerd.runc.v2 type=io.containerd.event.v1
[T0+dt]      [containerd][I] time="T0+dt"       level=info msg="loading plugin \"io.containerd.internal.v1.shutdown\"..." runtime=io.containerd.runc.v2 type=io.containerd.internal.v1
[T0+dt]      [containerd][I] time="T0+dt"       level=info msg="loading plugin \"io.containerd.ttrpc.v1.task\"..." runtime=io.containerd.runc.v2 type=io.containerd.ttrpc.v1
[T0+dt]      [containerd][I] time="T0+dt"       level=info msg="starting signal loop" namespace=moby path=/var/run/desktop-containerd/daemon/io.containerd.runtime.v2.task/moby/sha:a723... pid=17402 runtime=io.containerd.runc.v2
[T0+40s]     [containerd][I] time="T0+40s"      level=info msg="shim disconnected" id=sha:a723...
[T0+40s+dt]  [containerd][I] time="T0+40s+dt"   level=warning msg="cleaning up after shim disconnected" id=sha:a723... namespace=moby
[T0+40s+dt]  [containerd][I] time="T0+40s+dt"   level=info msg="cleaning up dead shim"
[T0+40s+dt]  [containerd][I] time="T0+40s+dt"   level=warning msg="cleanup warnings time=\"T0+40s+dt\" level=info msg=\"starting signal loop\" namespace=moby pid=17556 runtime=io.containerd.runc.v2\n"
[T0+45s]     [containerd][I] time="T0+45s"      level=info msg="loading plugin \"io.containerd.event.v1.publisher\"..." runtime=io.containerd.runc.v2 type=io.containerd.event.v1
[T0+45s+dt]  [containerd][I] time="T0+45s+dt"   level=info msg="loading plugin \"io.containerd.internal.v1.shutdown\"..." runtime=io.containerd.runc.v2 type=io.containerd.internal.v1
[T0+45s+dt]  [containerd][I] time="T0+45s+dt"   level=info msg="loading plugin \"io.containerd.ttrpc.v1.task\"..." runtime=io.containerd.runc.v2 type=io.containerd.ttrpc.v1
[T0+45s+dt]  [containerd][I] time="T0+45s+dt"   level=info msg="starting signal loop" namespace=moby path=/var/run/desktop-containerd/daemon/io.containerd.runtime.v2.task/moby/sha:f3d9... pid=17717 runtime=io.containerd.runc.v2
[T0+55s]     [containerd][I] time="T0+55s"      level=info msg="loading plugin \"io.containerd.event.v1.publisher\"..." runtime=io.containerd.runc.v2 type=io.containerd.event.v1
[T0+55s+dt]  [containerd][I] time="T0+55s+dt"   level=info msg="loading plugin \"io.containerd.internal.v1.shutdown\"..." runtime=io.containerd.runc.v2 type=io.containerd.internal.v1
[T0+55s+dt]  [containerd][I] time="T0+55s+dt"   level=info msg="loading plugin \"io.containerd.ttrpc.v1.task\"..." runtime=io.containerd.runc.v2 type=io.containerd.ttrpc.v1
[T0+55s+dt]  [containerd][I] time="T0+55s+dt"   level=info msg="starting signal loop" namespace=moby path=/var/run/desktop-containerd/daemon/io.containerd.runtime.v2.task/moby/sha:d6be... pid=17929 runtime=io.containerd.runc.v2
[T0+1m30s]   [containerd][I] time="T0+1m30s"    level=info msg="shim disconnected" id=sha:f3d9...
[T0+1m30s+dt][containerd][I] time="T0+1m30s+dt" level=warning msg="cleaning up after shim disconnected" id=sha:f3d9... namespace=moby
[T0+1m30s+dt][containerd][I] time="T0+1m30s+dt" level=info msg="cleaning up dead shim"
[T0+1m30s+dt][containerd][I] time="T0+1m30s+dt" level=warning msg="cleanup warnings time=\"T0+1m30s+dt\" level=info msg=\"starting signal loop\" namespace=moby pid=18062 runtime=io.containerd.runc.v2\n"
```

</details>

I don't know if this means anything, just want to get it out there.

**Comment by distortedsignal** [comment]
Hey Docker Team, let me know what I should be doing to help repro/troubleshoot this issue! :)

**Comment by distortedsignal** [comment]
Hey Docker Team, I'm still running into this issue regularly (not daily, but more often than weekly), and I was curious if there's an update or anything that I can do to help you repro this.

Thanks!

**Comment by distortedsignal** [comment]
Hey Docker Team, I've got more data!

It looks like I've got a hybrid 1-and-2 failure. Some containers have entered `Running` state, but get 0% CPU, while other containers are stuck in the `Created` state. Interesting!

The Diagnostic ID I generated from this failure was `EA6A9D0F-12E1-4F03-B4D4-915B28957161/20230922174453`. Let me know what else you need from me!

**Comment by distortedsignal** [comment]
If it helps to diagnose the issue, I'm trying to start docker containers from many OS processes. My process tree would look like this:

```
C---┬---┬---┬-...
|   |   |   | ...
W1  W2  W3  W4...
```

Where `C` is the controller process, and `W1` etc. are worker processes that can (serially) spawn several containers. So there will only ever be 1 container per process in the running state, but the containers will all be spawned by different PIDs. I don't know if this has any relevance to the problem, but it might.

**Comment by distortedsignal** [comment]
Hey docker, how much CPU are you using?

<img width="375" alt="Screenshot 2023-09-22 at 11 32 41 AM" src="https://github.com/docker/for-mac/assets/2186874/7636e083-72ac-452b-a24f-90221706187f">

**Comment by distortedsignal** [comment]
Still experiencing this issue. Still don't know how to help more than sending reports.

`EA6A9D0F-12E1-4F03-B4D4-915B28957161/20230928185019`

**Comment by distortedsignal** [comment]
Ok, did some poking around and I found the docker-diagnose util (on mac: `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose`) and it seems like this gives me more information about why this is failing. Before starting my process:

<details><summary>Working Docker Daemon</summary>

```
docker-diagnose check
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0038: is the connection to Docker working?
[PASS] DD0014: are the backend processes running?
[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[SKIP] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0032: do Docker networks overlap with host IPs?
No fatal errors detected.
```

</details>

After the process fails:

<details><summary>Failed Docker Daemon</summary>

```
docker-diagnose check
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[FAIL] DD0017: can a VM be started? vm has not started: vm has not started
[FAIL] DD0016: is the LinuxKit VM running? prereq failed: can a VM be started?
[FAIL] DD0011: are the LinuxKit services running? prereq failed: is the LinuxKit VM running?
[FAIL] DD0004: is the Docker engine running? prereq failed: are the LinuxKit services running?
[PASS] DD0015: are the binary symlinks installed?
[FAIL] DD0031: does the Docker API work? prereq failed: is the Docker engine running?
[PASS] DD0013: is the $PATH ok?
[FAIL] DD0003: is the Docker CLI working? prereq failed: is the Docker engine running?
[FAIL] DD0038: is the connection to Docker working? prereq failed: is the Docker engine running?
[FAIL] DD0014: are the backend processes running? prereq failed: is the LinuxKit VM running?
[FAIL] DD0007: is the backend responding? prereq failed: are the backend processes running?
[PASS] DD0008: is the native API responding?
[FAIL] DD0009: is the vpnkit API responding? prereq failed: are the backend processes running?
[FAIL] DD0010: is the Docker API proxy responding? prereq failed: are the backend processes running?
[FAIL] DD0030: is the image access management authorized? prereq failed: is the Docker engine running?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[WARN] DD0017: can a VM be started? vm has not started: vm has not started
[WARN] DD0016: is the LinuxKit VM running? prereq failed: can a VM be started?
[WARN] DD0011: are the LinuxKit services running? prereq failed: is the LinuxKit VM running?
[WARN] DD0004: is the Docker engine running? prereq failed: are the LinuxKit services running?
[PASS] DD0015: are the binary symlinks installed?
[WARN] DD0031: does the Docker API work? prereq failed: is the Docker engine running?
[WARN] DD0032: do Docker networks overlap with host IPs? prereq failed: does the Docker API work?

Please note the following 6 warnings:

1 : The check: can a VM be started?
    Produced the following warning: vm has not started: vm has not started

The Docker engine runs inside a Linux VM. Therefore we must be able to start Virtual Machines.

2 : The check: is the LinuxKit VM running?
    Produced the following warning: prereq failed: can a VM be started?

The Docker engine runs inside a Linux VM. Therefore the VM must be running.

3 : The check: are the LinuxKit services running?
    Produced the following warning: prereq failed: is the LinuxKit VM running?

The Docker engine runs inside a Linux VM as a service. Therefore the services must have started.

4 : The check: is the Docker engine running?
    Produced the following warning: prereq failed: are the LinuxKit services running?

The Docker engine manages all containers and images on the host. Check the dockerd.log to see why it failed to start.

5 : The check: does the Docker API work?
    Produced the following warning: prereq failed: is the Docker engine running?

If the Docker API is not available from the host then Docker Desktop will not work correctly.

6 : The check: do Docker networks overlap with host IPs?
    Produced the following warning: prereq failed: does the Docker API work?

If the subnet used by a Docker network overlaps with an IP used by the host, then containers
won't be able to contact the overlapping IP addresses.

Try configuring the IP address range used by networks: in your docker-compose.yml.
See https://docs.docker.com/compose/compose-file/compose-file-v2/#ipv4_address-ipv6_address


Please investigate the following 1 issue:

1 : The test: can a VM be started?
    Failed with: vm has not started: vm has not started

The Docker engine runs inside a Linux VM. Therefore we must be able to start Virtual Machines.
```

</details>

So it looks like something is preventing the Docker daemon from creating a certain number of VMs in a given time period? And then Docker hangs and dies because of that? I'm not 100% convinced by the Docker Diagnose tool, but it seems plausible.

**Comment by distortedsignal** [comment]
It looks like there's an update for me. I'm currently running a job, but when it's done, I'll update Docker and try again.

**Comment by bsousaa** [comment]
Please let us know how it goes with 4.24.

**Comment by ncouse** [comment]
@distortedsignal Note that that `diagnose check` command often gives the same failures for me (i.e. VM is not up), even though everything is OK. 


One interesting thing from the `diagnose check` command after update to 4.24, is that one of my networks was overlapping with the Docker VM IP subnet.
This network was created by `compose` without any explicit subnet specified. I just brought that compose project down and back up to solve.

**Comment by distortedsignal** [comment]
I'm not doing anything special with networking, so I kinda doubt that's the answer. Let me try with 4.24, and I'll get back to you.

**Comment by distortedsignal** [comment]
I'm not going to declare Can Not Reproduce (my soak test of "Let's run the prior job 10x in parallel" isn't quite done yet), but I'm going to whisper it... hesitantly. I'll let the job complete (probably sometime over the next 5 hours) and report back with the findings. Hopefully we can close this soon.

**Comment by distortedsignal** [comment]
I think I'm ready to say this is fixed in 4.23. I've been running 10 concurrent versions of the job that caused this to crash earlier and it seems significantly more stable. My CPU and SSD are weeping openly, but the daemon seems better.
